File: codes/from_python_to_go/closures.go
package main

import "fmt"

// intSeq return a function that returns a sequence of integers
func intSeq() func() int {
	i := 0
	return func() int {
		i += 1
		return i
	}
}

func main() {
	nextInt := intSeq()
	fmt.Println(nextInt())
	fmt.Println(nextInt())
	fmt.Println(nextInt())
}

File: codes/from_python_to_go/closures.py
#!/usr/bin/python3


def integer_sequence():
    '''Returns a sequence of integers'''
    i = 0
    def sequence():
        nonlocal i
        i += 1
        return i
    return sequence

next_int = integer_sequence()
print(next_int())
print(next_int())
print(next_int())
